src/Yesod/Auth/Simple.hs

```diff
@@ -419,13 +439,21 @@ postRegisterR = do
                     \ if the problem persists."
           redirectWithError registerR msg
     Just email' -> do
-      tp <- getRouteToParent
-      renderUrl <- getUrlRender
-      rawToken <- liftIO genToken
-      let url = renderUrl . tp . confirmTokenR $ encodeToken rawToken
-          hashed = hashAndEncodeToken rawToken
-      sendVerifyEmail (Email email') url hashed
-      redirect $ tp confirmationEmailSentR
+      getUserId email' >>= \case
```

Comment:
This logic may be too specific for yesod-auth-simple as a library, someone else might want to handle an existing email differently, but it makes sense in general.

Reply:
You're absolutely right. :100: Released it.

Perhaps we should add more features to the export to give users more flexibility in changing the logic.